### PR TITLE
crimson/os/alienstore: fix cpuset format of alienstore's threads

### PR DESCRIPTION
**Pull request by ljx023**
Fix cpuset format of  alienstore threads. Change it to cpuset(7) format.
Take cores "1,2,3,5,7,8" for example, previously it should be "1-4,5-6,7-9", now it is "1-3,5,7-8"

Signed-off-by: Jianxin Li <jianxin1.li@intel.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/alienstore/alien_store.cc [hunk 1]
```diff
@@ -619,9 +619,11 @@ std::vector<uint64_t> AlienStore::_parse_cpu_cores()
     std::string cpu;
     std::getline(cpu_stream, cpu, '-');
     uint64_t start_cpu = std::stoull(cpu);
+    uint64_t end_cpu = start_cpu;
     std::getline(cpu_stream, cpu, '-');
-    uint64_t end_cpu = std::stoull(cpu);
-    for (uint64_t i = start_cpu; i < end_cpu; i++) {
+    if(!cpu.empty())
+        end_cpu = std::stoull(cpu);
```
Comment (cyx1231st) [inline src/crimson/os/alienstore/alien_store.cc:625]:
```c++
if (!cpu.empty()) {
  end_cpu = std::stoull(cpu);
}
```

Comment (cyx1231st) [reply src/crimson/os/alienstore/alien_store.cc:625]: @ljx023 the white space is not consistent.

Comment (ljx023) [reply src/crimson/os/alienstore/alien_store.cc]: changed

**Review by cyx1231st — APPROVED** [review]

**Comment by cyx1231st** [comment]
Does it need to run the teuthology test suites before merge? @athanatos

**Review by xxhdx1985126 — APPROVED** [review]

File: src/crimson/os/alienstore/alien_store.cc [hunk 2]
```diff
@@ -619,9 +619,12 @@ std::vector<uint64_t> AlienStore::_parse_cpu_cores()
...
     std::getline(cpu_stream, cpu, '-');
     uint64_t start_cpu = std::stoull(cpu);
+    uint64_t end_cpu = start_cpu;
     std::getline(cpu_stream, cpu, '-');
-    uint64_t end_cpu = std::stoull(cpu);
-    for (uint64_t i = start_cpu; i < end_cpu; i++) {
+    if (!cpu.empty()) {
+      end_cpu = std::stoull(cpu);
+    }
+    for (uint64_t i = start_cpu; i <= end_cpu; i++) {
```
Comment (tchaikov) [inline src/crimson/os/alienstore/alien_store.cc:627]: i'd suggest use `seastar::resource::cpuset`. will come up with a patch shortly.

Comment (tchaikov) [reply src/crimson/os/alienstore/alien_store.cc:627]: #48716

Comment (ljx023) [reply src/crimson/os/alienstore/alien_store.cc:627]: > #48716

Well. I think this is a better implementation.

Comment (athanatos) [reply src/crimson/os/alienstore/alien_store.cc:627]: @cyx1231st We can merge this without running the teuthology suite.  The required lab quay instance is down and we haven't actually started requiring it yet, so I think we'll just delay that until the lab is back up.

**Comment by athanatos** [comment]
@cyx1231st We can merge this without running the teuthology suite. The required lab quay instance is down and we haven't actually started requiring it yet, so I think we'll just delay that until the lab is back up.

File: src/crimson/os/alienstore/alien_store.cc (hunk 2 above)
Comment (athanatos) [reply src/crimson/os/alienstore/alien_store.cc:627]: @ljx023 What's the disadvantage to using seastar's parsing code?

Comment (ljx023) [reply src/crimson/os/alienstore/alien_store.cc:627]: > @ljx023 What's the disadvantage to using seastar's parsing code?

I mean using seastar's parsing code is a better implementation. There is little difference between the two methods. Using seastar's parsing code does short the path.

**Comment by ljx023** [comment]
Close this PR due to a better implementation [#48716](https://github.com/ceph/ceph/pull/48716).

File: src/crimson/os/alienstore/alien_store.cc (hunk 2 above)
Comment (athanatos) [reply src/crimson/os/alienstore/alien_store.cc:627]: Oh, I misunderstood.  Agreed!

**Pull request closed without merging**